WordPress/src/main/java/org/wordpress/android/viewmodel/PluginBrowserViewModel.kt

```diff
@@ -0,0 +1,465 @@
+    val searchResults: LiveData<List<ImmutablePluginModel>>
+        get() = _searchResults
+
+    private val _featuredPluginsListStatus = MutableLiveData<PluginListStatus>()
+    val featuredPluginsListStatus: LiveData<PluginListStatus>
+        get() = _featuredPluginsListStatus
+
+    private val _newPluginsListStatus = MutableLiveData<PluginListStatus>()
+    val newPluginsListStatus: LiveData<PluginListStatus>
+        get() = _newPluginsListStatus
+
+    private val _popularPluginsListStatus = MutableLiveData<PluginListStatus>()
+    val popularPluginsListStatus: LiveData<PluginListStatus>
+        get() = _popularPluginsListStatus
+
+    private val _sitePluginsListStatus = MutableLiveData<PluginListStatus>()
+    val sitePluginsListStatus: LiveData<PluginListStatus>
+        get() = _sitePluginsListStatus
+
+    private val _searchPluginsListStatus = MutableLiveData<PluginListStatus>()
+    val searchPluginsListStatus: LiveData<PluginListStatus>
+        get() = _searchPluginsListStatus
+
+    private val _title = MutableLiveData<String>()
+    val title: LiveData<String>
+        get() = _title
+
+    var site: SiteModel? = null
+
+    var searchQuery: String by Delegates.observable("") {
+        _, oldValue, newValue ->
+        if (newValue != oldValue) {
+            submitSearch(newValue, true)
+        }
+    }
+
+    init {
+        mDispatcher.register(this)
+    }
+
+    override fun onCleared() {
+        mDispatcher.unregister(this)
+        super.onCleared()
+    }
+
+    fun writeToBundle(outState: Bundle) {
+        outState.putSerializable(WordPress.SITE, site)
+        outState.putString(KEY_SEARCH_QUERY, searchQuery)
+        outState.putString(KEY_TITLE, _title.value)
+    }
+
+    fun readFromBundle(savedInstanceState: Bundle) {
+        if (isStarted) {
+            // This was called due to a config change where the data survived, we don't need to
+            // read from the bundle
+            return
+        }
+        site = savedInstanceState.getSerializable(WordPress.SITE) as SiteModel
+        searchQuery = savedInstanceState.getString(KEY_SEARCH_QUERY)
+        setTitle(savedInstanceState.getString(KEY_TITLE))
+    }
+
+    fun start() {
+        if (isStarted) {
+            return
+        }
+        reloadPluginDirectory(PluginDirectoryType.FEATURED)
+        reloadPluginDirectory(PluginDirectoryType.NEW)
+        reloadPluginDirectory(PluginDirectoryType.POPULAR)
+        reloadPluginDirectory(PluginDirectoryType.SITE)
+
+        fetchPlugins(PluginListType.SITE, false)
+        fetchPlugins(PluginListType.FEATURED, false)
+        fetchPlugins(PluginListType.POPULAR, false)
+        fetchPlugins(PluginListType.NEW, false)
+
+        isStarted = true
+    }
+
+    // Site & WPOrg plugin management
+
+    private fun reloadPluginDirectory(directoryType: PluginDirectoryType) {
+        val pluginList = mPluginStore.getPluginDirectory(site!!, directoryType)
```

Comment:
Is there a possibility this method might get called before `site` is set? If so, just to be safe, you could wrap the content of the method in ```site?.let { ... }```.

Reply:
Fixed in 2ffd7aa99b9e61074188e9206983dc8cff8bbd14. I was hoping to remove the `reloadPluginDirectory` or change how it works in the next PR, that's why I skipped the `site!!`, but I think this is still a good change for now. I might not end up being able to remove it anyway.

I am surprised that we still need to call the `site!!` after a null check. I've noticed similar things in different places as well, I expected the compiler to know that it can't be `null` 🤔

Reply:
🤦‍♂️ 

I needed to use `it` after the `let`. I've made this change in my upcoming PR, so let's not add another commit just for that.